core/src/main/scala/tofu/syntax/feither.scala

```diff
@@ -0,0 +1,210 @@
+        case Right(right)   => f(right)
+        case left @ Left(_) => F.pure(left.rightCast)
+      }
+    }
+
+    def swapF(implicit F: Functor[F]): F[Either[R, L]] = {
+      F.map(e)(_.swap)
+    }
+
+    def ensure[L1 >: L](f: R => Boolean, err: => L1)(implicit F: Functor[F]): F[Either[L1, R]] = {
+      flatMapIn(right => Either.cond(f(right), right, err))
+    }
+
+    def ensureF[L1 >: L](f: R => F[Boolean], err: => F[L1])(implicit F: Monad[F]): F[Either[L1, R]] = {
+      flatMapF(right => F.flatMap(f(right))(p => Either.condF(p, F.pure(right), err)))
+    }
+
+    def traverseF[G[_]: Applicative, R1](f: R => G[R1])(implicit F: Functor[F]): F[G[Either[L, R1]]] = {
+      F.map(e)(_.traverse(f))
+    }
+
+    def traverseAll[G[_]: Applicative, R1](f: R => G[R1])(implicit F: Traverse[F]): G[F[Either[L, R1]]] = {
+      F.traverse(e)(_.traverse(f))
+    }
+
+    def leftTraverseF[G[_], L1](f: L => G[L1])(implicit F: Functor[F], G: Applicative[G]): F[G[Either[L1, R]]] = {
+      F.map(e) {
+        case Left(left)   => G.map(f(left))(_.asLeft)
+        case r @ Right(_) => G.pure(r.leftCast)
+      }
+    }
+
+    def leftTraverseAll[G[_], L1](f: L => G[L1])(implicit F: Traverse[F], G: Applicative[G]): G[F[Either[L1, R]]] = {
+      F.traverse(e) {
+        case Left(left)   => G.map(f(left))(_.asLeft)
+        case r @ Right(_) => G.pure(r.leftCast)
+      }
+    }
+
+    def productF[L1 >: L, R1](eb: => F[Either[L1, R1]])(implicit F: Monad[F]): F[Either[L1, (R, R1)]] = {
+      F.flatMap(e) {
+        case Right(r) =>
+          F.flatMap(eb) {
+            case Right(r1)      => (r, r1).asRightF
+            case left @ Left(_) => F.pure(left.rightCast)
+          }
+        case left @ Left(_) => F.pure(left.rightCast)
+      }
+    }
+
+    def productRF[L1 >: L, R1](eb: => F[Either[L1, R1]])(implicit F: Monad[F]): F[Either[L1, R1]] = {
+      F.map(productF(eb))(_.map(_._2))
+    }
+
+    def productLF[L1 >: L, R1](eb: => F[Either[L1, R1]])(implicit F: Monad[F]): F[Either[L1, R]] = {
+      F.map(productF(eb))(_.map(_._1))
+    }
+
+    def apF[L1 >: L, R1, Z](
+        eb: => F[Either[L1, R1]]
+    )(implicit F: Monad[F], ev: R <:< (R1 => Z)): F[Either[L1, Z]] = {
+      F.flatMap(e) {
+        case Right(r) =>
+          F.flatMap(eb) {
+            case Right(r1)      => r(r1).asRightF
+            case left @ Left(_) => F.pure(left.rightCast)
+          }
+        case left @ Left(_) => F.pure(left.rightCast)
+      }
+    }
+
+    def map2F[L1 >: L, R1, Z](
+        eb: => F[Either[L1, R1]]
+    )(f: (R, R1) => Z)(implicit F: Monad[F]): F[Either[L1, Z]] = {
+      F.map(productF(eb))(_.map { case (r, r1) => f(r, r1) })
+    }
+
+    def flatMap2F[L1 >: L, R1, Z](
+        eb: => F[Either[L1, R1]]
+    )(f: (R, R1) => F[Z])(implicit F: Monad[F]): F[Either[L1, Z]] = {
+      F.flatMap(productF(eb))(_.traverse { case (r, r1) => f(r, r1) })
+    }
+
+    def mapK[G[_]](implicit F: ~>[F, G]): G[Either[L, R]] = {
```

Comment:
I don't think that's a good idea to make natural transformation implicit.
Do we really need mapK and liftTo methods here?

Reply:
At the last minute i became doubted about it but decided to left it. So, if it doesn't looks good enough i can cut it.